### PR TITLE
fix rediscluster kubectl plugin

### PR DESCRIPTION
**Pull request by clamoriniere1A**
fixes: #14

add missing client-go import in order to support auth provider in the
kubectl rediscluster plugin.
Also improve the way the plugin process retrieve the kubeconfig file
path.

### CODE REVIEW

File: cmd/kubectl-plugin/main.go [hunk 1]
```diff
@@ -35,44 +34,40 @@ func main() {
...
 		glog.Fatalf("Unable to init redis.clientset from kubeconfig:%v", err)
 	}
 
-	var rcs *v1.RedisClusterList
+	rcs := &v1.RedisClusterList{}
 	if clusterName == "" {
 		rcs, err = redisClient.RedisoperatorV1().RedisClusters(namespace).List(meta_v1.ListOptions{})
 		if err != nil {
-			glog.Fatalf("unable to list rediscluster err:%v", err)
+			glog.Fatalf("unable to list rediscluster:%s err:%v", err)
```
Comment (sdminonne) [inline cmd/kubectl-plugin/main.go]: miss rediscluter name?

Comment (sdminonne) [reply cmd/kubectl-plugin/main.go]: or namespace... not sure

File: cmd/kubectl-plugin/main.go [hunk 2]
```diff
@@ -35,44 +34,40 @@ func main() {
...
 	} else {
-		rcs = &v1.RedisClusterList{}
 		rc, err := redisClient.RedisoperatorV1().RedisClusters(namespace).Get(clusterName, meta_v1.GetOptions{})
-		if err != nil {
-			glog.Fatalf("unable to list rediscluster err:%v", err)
+		if err == nil && rc != nil {
+			rcs.Items = append(rcs.Items, *rc)
+		}
+		if err != nil && !apierrors.IsNotFound(err) {
+			glog.Fatalf("unable to get rediscluster '%s' err:%v", clusterName, err)
```
Comment (sdminonne) [inline cmd/kubectl-plugin/main.go]: nit
~~`glog.Fatalf("unable to get rediscluster '%s' err:%v", clusterName, err)`~~ glog.Fatalf("unable to get rediscluster %q: %v", clusterName, err)
but it's a matter of taste... so

File: cmd/kubectl-plugin/main.go [hunk 3]
```diff
@@ -152,3 +147,59 @@ func buildReplicationStatus(rc *v1.RedisCluster) string {
...
+	}
+
+	var cfg clientcmd.ClientConfig
+	context := os.Getenv("KUBECTL_PLUGINS_GLOBAL_FLAG_CONTEXT")
+	if len(context) > 0 {
+		rules := clientcmd.NewDefaultClientConfigLoadingRules()
+		cfg = clientcmd.NewNonInteractiveClientConfig(*credentials, context, overrides, rules)
+	} else {
+		cfg = clientcmd.NewDefaultClientConfig(*credentials, overrides)
+	}
```
Comment (sdminonne) [inline cmd/kubectl-plugin/main.go]: NIT: I would rearrage to remove the `else`

File: cmd/kubectl-plugin/main.go (hunk 1 above)
Comment (clamoriniere1A) [reply cmd/kubectl-plugin/main.go]: thanks for your comments: new PR#16 to fix it all

**Pull request merged**